server/mailet/integration-testing/src/test/java/org/apache/james/mailets/SubAddressingTest.java

```diff
@@ -100,10 +101,62 @@ void subAddressedEmailShouldBeDeliveredInINBOXWhenSpecifiedFolderDoesNotExist(@T
 
         // do not create mailbox
 
-        sendSubAddressedMail();
+        sendSubAddressedMail(TARGETED_MAILBOX);
         awaitSubAddressedMail(MailboxConstants.INBOX);
     }
 
+    @Test
+    void subAddressedEmailShouldBeDeliveredInSpecifiedFolderWhenItExistsInUpperCase(@TempDir File temporaryFolder) throws Exception {
+        setup(temporaryFolder);
+
+        // create mailbox
+        testIMAPClient.sendCommand("CREATE " + TARGETED_MAILBOX_UPPER);
+        testIMAPClient.sendCommand("SETACL " + TARGETED_MAILBOX_UPPER + " " + "anyone" + " p");
+
+        sendSubAddressedMail(TARGETED_MAILBOX);
+        awaitSubAddressedMail(TARGETED_MAILBOX_UPPER);
+    }
+
+    @Test
+    void subAddressedEmailShouldBeDeliveredInSpecifiedFolderWhenItExistsInLowerCase(@TempDir File temporaryFolder) throws Exception {
+        setup(temporaryFolder);
+
+        // create mailbox
+        testIMAPClient.sendCommand("CREATE " + TARGETED_MAILBOX);
+        testIMAPClient.sendCommand("SETACL " + TARGETED_MAILBOX + " " + "anyone" + " p");
+
+        sendSubAddressedMail(TARGETED_MAILBOX_UPPER);
+        awaitSubAddressedMail(TARGETED_MAILBOX);
+    }
+
+    @Test
+    void subAddressedEmailShouldBeDeliveredInSpecifiedFolderWithCorrectLowerCaseWhenSeveralCasesExist(@TempDir File temporaryFolder) throws Exception {
+        setup(temporaryFolder);
+
+        // create mailboxes
+        testIMAPClient.sendCommand("CREATE " + TARGETED_MAILBOX);
+        testIMAPClient.sendCommand("SETACL " + TARGETED_MAILBOX + " " + "anyone" + " p");
+        testIMAPClient.sendCommand("CREATE " + TARGETED_MAILBOX_UPPER);
+        testIMAPClient.sendCommand("SETACL " + TARGETED_MAILBOX_UPPER + " " + "anyone" + " p");
+
+        sendSubAddressedMail(TARGETED_MAILBOX);
+        awaitSubAddressedMail(TARGETED_MAILBOX);
+    }
+
+    @Test
+    void subAddressedEmailShouldBeDeliveredInSpecifiedFolderWithCorrectUpperCaseWhenSeveralCasesExist(@TempDir File temporaryFolder) throws Exception {
+        setup(temporaryFolder);
+
+        // create mailboxes
+        testIMAPClient.sendCommand("CREATE " + TARGETED_MAILBOX);
+        testIMAPClient.sendCommand("SETACL " + TARGETED_MAILBOX + " " + "anyone" + " p");
+        testIMAPClient.sendCommand("CREATE " + TARGETED_MAILBOX_UPPER);
+        testIMAPClient.sendCommand("SETACL " + TARGETED_MAILBOX_UPPER + " " + "anyone" + " p");
```

Comment:
```suggestion
        testIMAPClient.sendCommand("CREATE " + TARGETED_MAILBOX);
        testIMAPClient.sendCommand("SETACL " + TARGETED_MAILBOX + " " + "anyone" + " p");
        testIMAPClient.sendCommand("CREATE " + TARGETED_MAILBOX_UPPER);
```

Then what happens?

Reply:
the desired outcome was already correctly achieved, but i added a test for that specific case to make it clear :)